lib/srv/db/postgres/users.go

```diff
@@ -43,27 +45,25 @@ func (e *Engine) ActivateUser(ctx context.Context, sessionCtx *common.Session) e
 	// We could call this once when the database is being initialized but
 	// doing it here has a nice "self-healing" property in case the Teleport
 	// bookkeeping group or stored procedures get deleted or changed offband.
-	err = e.initAutoUsers(ctx, conn)
+	err = e.initAutoUsers(ctx, sessionCtx, conn)
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
-	roles := sessionCtx.DatabaseRoles
-	if sessionCtx.Database.IsRDS() {
-		roles = append(roles, "rds_iam")
+	roles, err := prepareRoles(sessionCtx)
+	if err != nil {
+		return trace.Wrap(err)
 	}
 
 	e.Log.Infof("Activating PostgreSQL user %q with roles %v.", sessionCtx.DatabaseUser, roles)
 
 	_, err = conn.Exec(ctx, activateQuery, sessionCtx.DatabaseUser, roles)
-	if err != nil {
-		if strings.Contains(err.Error(), "already exists") {
-			return trace.AlreadyExists("user %q already exists in this PostgreSQL database and is not managed by Teleport", sessionCtx.DatabaseUser)
-		}
-		return trace.Wrap(err)
+	if err == nil {
+		return nil
 	}
 
-	return nil
+	e.Log.Debugf("Call teleport_activate_user failed: %v", err)
+	return trace.Wrap(convertActivateError(sessionCtx, err))
```

Comment:
This inverted error handling tripped me up, I think we could have a standard `err != nil` branch here?

Reply:
lol. sure, will do